crates/saya/core/src/prover/stone_image.rs

```diff
@@ -1,26 +1,40 @@
+use std::fs::File;
 use std::process::Stdio;
 
 use anyhow::{bail, Context};
 use async_trait::async_trait;
 use tokio::io::{AsyncBufReadExt, AsyncReadExt, AsyncWriteExt, BufReader};
 use tokio::process::Command;
 use tokio::sync::OnceCell;
-use tracing::warn;
+use tracing::trace;
 
-use super::{ProverClient, ProverIdentifier};
+use super::{ProveProgram, ProverClient, ProverIdentifier};
+use crate::prover::loader::prepare_input_cairo0;
+use crate::LOG_TARGET;
+
+#[cfg(feature = "cairo1differ")]
+const PROVER_IMAGE: &str = "neotheprogramist/stone-cairo:latest";
+#[cfg(not(feature = "cairo1differ"))]
+const PROVER_IMAGE: &str = "neotheprogramist/stone-cairo0:latest";
 
 #[derive(Clone)]
 pub struct StoneProver(pub String);
 
-pub async fn prove_stone(input: String) -> anyhow::Result<String> {
+pub async fn prove_stone(input: String, prove_program: ProveProgram) -> anyhow::Result<String> {
     let prover = StoneProver::new().await?;
-    prover.prove(input).await
-}
 
-pub async fn local_verify(input: String) -> anyhow::Result<String> {
-    let prover = StoneProver::new().await?;
-    prover.local_verify(input).await?;
-    Ok(String::from("ok"))
+    if cfg!(feature = "cairo1differ") {
+        unimplemented!("Cairo1 prover for `StoneProver` not supported yet.")
+    } else {
+        trace!(target: LOG_TARGET, "Proving with cairo0.");
```

Comment:
Same here, if we have a way to effectively identify what's being proved, could be awesome.

Reply:
Actually, as language of proof does not change for the duration of running `saya`, it is enough to print it once.
Log about which block is being proven is printed before execution of this function.

Reply:
So changed it to print which language the program is using only once.